XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,99 @@
+---
+xip: 10
+title: Brand info schema in conversation metadata
+description: Schema to enable standardized sharing of brand information as it relates to conversationId
+author: Yash Lunagaria (@yash-luna)
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-01-31
+---
+
+## Abstract
+
+This XIP proposes a schema for sharing brand information as it relates to `conversationId`. A standard way to represent this information in conversation `metadata` will enable different front-ends to surface the context captured by `conversationId` in a consistent, user-friendly manner while still affording clients design and UI flexibility.
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways and therefore other apps may not know how to render the `conversationId` appropriately within the UI to reflect the corresponding brand. We propose a schema to standardize the use of conversation `metadata` for the purpose of client-side brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `brandInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: 'mydomain.xyz/abc/qrs',
+    metadata: {
+        brandInfo: {
+            displayName: 'My Company',
+            profileImage: 'mydomain.xyz/assets/myimage.png,
+            primaryColor: '#ffffff',
+        }
+    }
+}
+```
+
+Example `brandInfo` implementation for a chat app named Galaxy
+
+```json
+{
+    conversationId: 'galaxy.chat/dm/uniqueIdentifier',
+    metadata: {
+        brandInfo: {
+            displayName: 'Galaxy',
+            profileImage: 'galaxychat.com/assets/logo.png,
+            primaryColor: '#6865B8',
+        }
+    }
+}
+```
+
+The `profileImage` file is encouraged to meet the following criteria:
```

Comment:
Why SHOULD rather than MUST?

If this is a MUST, then we'd expect clients to validate and reject images that fail to meet the criteria. If a SHOULD, then they can decide at-will how to deal with non-conforming image.

Reply:
It MUST!